source/getting-started/tutorials/sensors/index.md

```diff
@@ -0,0 +1,283 @@
+Loop Num:	48373 samples
+Loop Min:	80.46 usec
+Loop Max:	119.44 usec
+Loop Mean:	100.00 usec
+Loop Var:	0.24 usec
+Run Num:	48624 samples
+Run Min:	9.85 usec
+Run Max:	17.02 usec
+Run Mean:	10.25 usec
+Run Var:	1.40 usec
+```
+
+The loop time has returned to 100.00 usec. The timing manager is not slowing down the rate of the control task anymore.
+
+But the Run time has increased significantly. Why is that?
+
+The answer is that I have no idea. Ask Patrick
+
+## Experiment 3 - Changing PWM frequency
+
+Another way to impact loop time is to change the PWM frequency. Let's return to the situation with a User Event Ratio of 10, but this time modify the PWM ratio from 100KHz to 50KHz. We can do this by adding the code `pwm_set_switching_freq(50000)` to our init function (remember to `#include "drv/pwm.h"` at the top of the file).
+
+`app_controller.c`:
+```
+#include "drv/pwm.h"
+
+void app_controller_init(void)
+{
+    // Enable data sampling for ADC
+    timing_manager_enable_sensor(ADC);
+    // set User Event Ratio
+    timing_manager_set_ratio(10);
+    // set PWM frequency
+    pwm_set_switching_freq(50000);
+    // Register "ctrl" command with system
+    cmd_controller_register();
+}
+```
+
+Now we're sampling sensors every `10` PWM cycles, but each PWM cycle takes twice as long as it used to. This means that our sensors are only sampled every 200us again, which will slow down our control task rate!
+
+![](images/tmVSI50.svg)
+
+Rebuild and run the new program, and use the command `ctrl stats print` to view the loop time (after doing `ctrl init`).
+
+```
+Task Stats:
+Loop Num:	10847 samples
+Loop Min:	193.82 usec
+Loop Max:	206.22 usec
+Loop Mean:	200.00 usec
+Loop Var:	0.04 usec
+Run Num:	10973 samples
+Run Min:	3.25 usec
+Run Max:	4.24 usec
+Run Mean:	3.39 usec
+Run Var:	0.00 usec
+```
+
+Indeed our loop mean is back to 200us.
+
+We can fix this by adjusting the timing manager's ratio from 10 down to 5.
+
+```
+void app_controller_init(void)
+{
+    // Enable data sampling for ADC
+    timing_manager_enable_sensor(ADC);
+    // set User Event Ratio
+    timing_manager_set_ratio(5);
+    // set PWM frequency
+    pwm_set_switching_freq(50000);
+    // Register "ctrl" command with system
+    cmd_controller_register();
+}
+```
+
+Now we're sampling sensors every `5` PWM cycles. Since each PWM cycle takes 20us, that means our sensor sampling and control task will run every 100us once again.
+
+![](images/tmVSI550.svg)
+
+Rebuild and run the new program, and use the command `ctrl stats print` to view the loop time (after doing `ctrl init`).
+
+```
+Task Stats:
+Loop Num:	27497 samples
+Loop Min:	93.69 usec
+Loop Max:	106.35 usec
+Loop Mean:	100.00 usec
+Loop Var:	0.02 usec
+Run Num:	27748 samples
+Run Min:	3.25 usec
+Run Max:	3.84 usec
+Run Mean:	3.39 usec
+Run Var:	0.00 usec
+```
+
+## TODO
+
+ISR Generation mode: Legacy vs New Mode
+- when it makes an impact
+- how to use the `is_sensor_done` in a loop to poll/stall in Legacy Mode
```

Comment:
Will this be finished as part of this PR?

Reply:
Probably not, unless you have any suggestions on how it can be incorporated